pika/heartbeat.py

```diff
@@ -165,23 +169,40 @@ def _send_heartbeat_frame(self):
             self._new_heartbeat_frame())
         self._heartbeat_frames_sent += 1
 
-    def _setup_timer(self):
-        """Use the connection objects delayed_call function which is
-        implemented by the Adapter for calling the check_heartbeats function
-        every interval seconds.
+    def _setup_timers(self):
+        """Use the connection objects _adapter_add_timeout function which is
+        implemented by the Adapter for calling our send and check functions
+        after interval seconds.
+
+        """
+        self._setup_send_timer()
+        self._setup_check_timer()
+
+    def _setup_send_timer(self): # pylint: disable=C0111
+        self._send_timer = self._connection._adapter_add_timeout(  # pylint: disable=W0212
+            self._send_interval,
+            self.send_heartbeat)
```

Comment:
send_hearbeat and check_heartbeat should be private methods with leading underscore in the names.

Reply:
OK, `send_and_check` used to be public, so I kept it that way.